src/m365/aad/commands/user/user-hibp.ts

```diff
@@ -0,0 +1,100 @@
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import Command from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import commands from '../../commands';
+import Utils from '../../../../Utils';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  userName: string;
+  apiKey: string;
+  domain?: string
+}
+
+class HIBPCommand extends Command {
+  public get name(): string {
+    return commands.USER_HIBP;
+  }
+
+  static readonly HIBP_HOST = 'https://haveibeenpwned.com/api/v3';
+
+  public get description(): string {
+    return 'Allows you to retrieve all or accounts from specified domain that have been pwned with the specified username';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+
+    let url: string = `https://haveibeenpwned.com/api/v3/breachedaccount/${args.options.userName}`;
+    
+    // Filters the result set to only breaches against the domain specified.
+    if (args.options.domain) {
+      url += `?domain=${args.options.domain}`;
+    }
+
+    const requestOptions: any = {
+      url: url,
+      headers: {
+        'accept': 'application/json',
+        'hibp-api-key': args.options.apiKey,
+        'x-anonymous': true
+      },
+      responseType: 'json'
+    };
+
+    request
+      .get(requestOptions)
+      .then((res: any): void => {
+        logger.log(res);
+        cb();
+      }, (rawRes: any): void => {
+        if (rawRes.response !== undefined && rawRes.response.status === 404) {
+          logger.log("\nGood news — no pwnage found!\n");
```

Comment:
Let's include this response in verbose and text mode only. Otherwise, it could get hard to handle in JSON output mode where the script is expecting a JSON array and we're passing a string.